docs/adapters.rst

```diff
@@ -460,6 +460,35 @@ Or via query parameters:
 
 Note that if passing the headers via query parameters the dictionary should be serialized using `RISON <https://pypi.org/project/prison/>`_.
 
+To ensure secret header values aren't sent to the wrong URL, you may wish to specify that certain request headers should only be sent to certain origins. You may do so via the ``request_headers_by_origin`` keyword argument, which should contain a map from origins (scheme + domain + port) to request header objects. Note that URI parts other than scheme, domain, or port will be stripped out of this config value before comparing.
+
+.. code-block:: python
+
+    from shillelagh.backends.apsw.db import connect
+
+    connection = connect(
+        ":memory:",
+        adapter_kwargs={
+            "genericjsonapi": {
+                "request_headers_by_origin": {
```

Comment:
I'm wondering if this should just be `origins` or `config_by_origin` or something like that, and then the value can be a map from origin to a dict of config values.

So e.g. this would become:

```
                 "origin_configs": {
                    "https://api.example.com": {
                        "request_headers": {
                            "X-Auth-Token": "SECRET",
                        },
                    },
                },
```

That then leaves room for per-origin assignment of other config keys if more get added down the road -- e.g. `username`/`password` basic auth credentials, or allowing/disallowing redirects, etc.